migrations/1680127440523-add-guardian-ideal.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { ObjectId } = require('mongodb')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { runMigration } = require('../utils/mongodb')
+
+module.exports.up = runMigration(async (db) => {
+  // If the first item is FeaturedShortcuts, and the second item in a
+  // is not Guardian Ideal, add it as the second item in a user's MySpace.
+  // `default: true` lets us know we've added it as a default and not the user.
+  await db.collection('users').updateMany(
+    {
+      'mySpace.0.type': {
+        $eq: 'FeaturedShortcuts',
+      },
+      'mySpace.1.type': {
+        $ne: 'GuardianIdeal',
+      },
+    },
+    {
+      $push: {
+        mySpace: {
+          $each: [
+            {
+              _id: new ObjectId(),
+              title: 'Guardian Ideal',
+              type: 'GuardianIdeal',
+              default: true,
+            },
+          ],
+          $position: 1,
+        },
+      },
+    }
+  )
+
+  // If the first item is not FeaturedShortcuts or GuardianIdeal,
+  // add GuardianIdeal as the first item in a user's MySpace.
+  await db.collection('users').updateMany(
+    {
+      'mySpace.0.type': {
+        //$nin checks that the value is not in the array of values provided
+        $nin: ['GuardianIdeal', 'FeaturedShortcuts'],
```

Comment:
Question: Is it worth it to add both in this migration if the user doesn't have either? Now that we add it for new users we'll need to have a migration to add `FeaturedShortcuts` again, but I'm fine if we want to keep that separate from this one.

Reply:
let's keep that as a separate PR, just to keep the migrations a bit simpler. i feel like this PR has gotten a bit more complex than the original story as is.